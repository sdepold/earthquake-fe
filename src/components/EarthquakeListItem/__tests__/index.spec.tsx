import { render } from "@testing-library/react";
import { EarthquakeListItem, EarthquakeListItemProps } from "..";

jest.mock("../../Map", () => ({
  __esModule: true,
  default: () => <div>Map</div>,
}));

describe("EarthquakeListItem", () => {
  it("should match the expected HTML structure", () => {
    const props: EarthquakeListItemProps = {
      earthquake: {
        magnitude: 3.58,
        place: "2 km NE of Newport Beach, CA",
        date: new Date(1717717944590),
        url: "https://earthquake.usgs.gov/earthquakes/eventpage/ci40616303",
        title: "M 3.6 - 2 km NE of Newport Beach, CA",
        lat: 33.6311667,
        long: -117.9098333,
        height: 12.08,
        id: "ci40616303",
      },
    };

    const { container } = render(<EarthquakeListItem {...props} />);
    expect(container).toMatchSnapshot();
  });
});
