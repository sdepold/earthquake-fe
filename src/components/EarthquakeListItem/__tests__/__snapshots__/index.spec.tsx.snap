// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarthquakeListItem should match the expected HTML structure 1`] = `
<div>
  <div
    class="earthquake-list-item"
  >
    <div>
      Map
    </div>
    <div>
      <h2>
        2 km NE of Newport Beach, CA
      </h2>
      <p>
        6/7/2024
      </p>
    </div>
    <div
      class="magnitude"
    >
      <span>
        4
      </span>
    </div>
  </div>
</div>
`;
